pkg/resource/generator.go

```diff
@@ -284,6 +286,41 @@ func (g *Generator) Remove(cr *imageregistryv1.Config, modified *bool) error {
 		glog.Infof("object %s deleted", Name(gen))
 		*modified = true
 	}
+	rclient, err := regopset.NewForConfig(g.kubeconfig)
+	if err != nil {
+		return err
+	}
+	driver, err := storage.NewDriver(cr.ObjectMeta.Name, cr.ObjectMeta.Namespace, &cr.Status.Storage)
+	if err != nil {
+		return err
+	}
+
+	o := cr
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if o == nil {
+			o, err = rclient.Configs().Get(cr.Name, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get %s, retrying: %v", util.ObjectInfo(cr), err)
+			}
+			driver.UpdateFromStorage(o.Status.Storage)
+		}
+		modified := false
+
+		if err := driver.RemoveStorage(cr, &modified); err != nil {
+			o = nil
```

Comment:
i'm not seeing the point of this assignment.

Reply:
updated